### DISCUSSION TITLE
Failed to create wallet from private key. Private key is invalid hex: Odd number of digits

### DISCUSSION THREAD

**Discussion by satyam455** [Cyfrin/foundry-full-course-cu#50] [Q&A] [answered]
i entered this cmd , ```cast send 0x5fbdb2315678afecb367f032d93f642f64180aa3 "store(uint256)" 123 --rpc-url $RPC_URL --private-key $PRIVATE_KEY``` 

and it is  showing me this error: 
error: Failed to create wallet from private key. Private key is invalid hex: Odd number of digits
Context:
- Odd number of digits

**Comment by cromewar** [comment]
Hello @satyam455 
Make sure the private key is being loaded on the terminal, as this is going to try to use it from env variables, you can do this on your terminal:

`export PRIVATE_KEY yourKey` 

and they try using the command again, if works means that the Private key wasn't loaded correctly.

**Reply by satyam455** [reply]
ok

**Comment by NokivaF** [comment, marked as answer]
> i entered this cmd , `cast send 0x5fbdb2315678afecb367f032d93f642f64180aa3 "store(uint256)" 123 --rpc-url $RPC_URL --private-key $PRIVATE_KEY`
> 
> and it is showing me this error: error: Failed to create wallet from private key. Private key is invalid hex: Odd number of digits Context:
> 
> * Odd number of digits

I've already opened a thread treating this issue.  Always try to make a quick scan of the issues, they might have the answer your are looking for. [This](https://github.com/Cyfrin/foundry-full-course-f23/discussions/24#discussion-5279889) is the thread.
